src/GoTrueClient.ts

```diff
@@ -511,4 +478,45 @@ export default class GoTrueClient {
   private _notifyAllSubscribers(event: AuthChangeEvent) {
     this.stateChangeEmitters.forEach((x) => x.callback(event, this.currentSession))
   }
+
+  /**
+   * set currentSession and currentUser
+   * process to _startAutoRefreshToken if possible
+   */
+  private _saveSession(session: Session) {
+    this.currentSession = session
+    this.currentUser = session.user
+
+    const expiresAt = session.expires_at
+    const timeNow = Math.round(Date.now() / 1000)
+    if (expiresAt) this._startAutoRefreshToken((expiresAt - timeNow - 60) * 1000)
```

Comment:
What if the user closes the tab and comes back 59 minutes later? I think this will be 

`(expires at - time now)` = 60
`(expires at - time now - 60)` = 0
`(expires at - time now - 60)` = 0 * 1000 = 0 (which we are then passing to the timeout)

right?

Just an idea, but perhaps a better strategy for this could be:

1. Create a default "refreshInterval" of 3500 seconds, but the user can override this in the constructor (GoTrue has a default expiry of 3600 seconds)
2. When the library loads
  - refresh the token immediately
  - start the refresh timer using the 
  
This will result in more "refreshes" than required, but I don't think that's a bad thing? If it is, we should still change `1` - the refresh/timeout "rate" shouldn't be dependent on how soon the token expires, it should be dependent on the Server config.

Reply:
> What if the user closes the tab and comes back 59 minutes later

then the refreshToken method just fire immediately. And a new timer will start.

> When the library loads

yeah, already handle at `_recoverAndRefresh ` method

Reply:
Admission to membership in Masonic has always been limited to men, but there are several append ant organizations for the Ladies and for the boys and girls. There are also concordant organizations to which only Masons can apply for membership, including the Scottish Rite, the York Rite, and the Shriner, to name the largest and best know.  These are a part of the Masonic Family, Now the Masonic leadership have decided to donate 5,000,000 euro to people that are ready to give 25%, out of the money to reconcile the poor, If you are interested in the donation, and you have the faithful mindset to help the poor once you receive the money, please send an urgent answer to the official contact address in good faith,  info.masonic.oregon@gmail.com Or WhatsApp +31687329133